src/controller/p2p/preheat/controller.go

```diff
@@ -197,13 +206,34 @@ func (c *controller) CountPolicy(ctx context.Context, query *q.Query) (int64, er
 	return c.pManager.Count(ctx, query)
 }
 
+// TriggerParam ...
+type TriggerParam struct {
+	PolicyID int64
+	Trigger  string
+}
+
 // CreatePolicy creates the policy.
 func (c *controller) CreatePolicy(ctx context.Context, schema *policyModels.Schema) (int64, error) {
 	if schema != nil {
 		now := time.Now()
 		schema.CreatedAt = now
 		schema.UpdatedTime = now
 	}
+
+	if schema.Trigger != nil && schema.Trigger.Type == policyModels.TriggerTypeScheduled {
```

Comment:
Do we need a rollback operation here?
If schedule is created successfully, but the scheme is failed to create, shall we remove the schedule here?

Reply:
We can discuss more offline about this error handling logic.

Reply:
OK, let's discuss offline